packages/astro/src/core/build/index.ts

```diff
@@ -62,8 +64,7 @@ class AstroBuilder {
         {
           mode: this.mode,
           server: {
-            hmr: { overlay: false },
-            middlewareMode: 'ssr',
```

Comment:
This is needed so that we can cal `createServer()`. I didn't notice a difference without it.

Reply:
Yeah I’m not sure what all this does either. I was under the impression that it would skip a lot of setup for the normal dev server, possibly improving perf. But I’m not aware of any downsides to removing this.